modules/http-api/src/main/java/org/mule/runtime/http/api/sse/SseRequestContext.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.http.api.sse;
+
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface SseRequestContext {
+
+  HttpRequest getRequest();
+
+  void setClientId(String uuid);
+
+  CompletableFuture<Void> sendResponse(HttpResponse response);
```

Comment:
should this take a ResponseBuilder instead? Here's my thinking:

One side, there are a set of default headers you probably want to apply, unless the consumer specifically sets otherwise. So you can receive a builder and add default headers that don't contradict what the consumer set

Reply:
Agreed that we will change it to reject(statusCode, reasonPhrase)